include/deal.II/base/qprojector.h

```diff
@@ -22,12 +22,13 @@
 #include <deal.II/base/geometry_info.h>
 #include <deal.II/base/quadrature.h>
 
-#include <deal.II/grid/reference_cell.h>
-
 #include <deal.II/hp/q_collection.h>
 
 DEAL_II_NAMESPACE_OPEN
 
+#ifndef DOXYGEN
+class ReferenceCell;
+#endif
```

Comment:
I assume this is the reason why you need to switch all those appearances from pass-by-value to pass-by-reference, even though the actual class only contains a single `unit_8`? I agree with this change.

Reply:
Yup! `reference_cell.h` is a surprisingly large header to compile.